src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -383,16 +383,24 @@ export class ConnectionDialogService implements IConnectionDialogService {
 
 	private async updateModelServerCapabilities(model: IConnectionProfile) {
 		this._model = await this.createModel(model);
-		if (this._model.providerName) {
+		if (this._model?.providerName) {
 			this._currentProviderType = this._model.providerName;
 			if (this._connectionDialog) {
 				this._connectionDialog.updateProvider(this._providerNameToDisplayNameMap[this._currentProviderType]);
 			}
 		}
 	}
 
-	private async createModel(model: IConnectionProfile): Promise<ConnectionProfile> {
+	private async createModel(model: IConnectionProfile): Promise<ConnectionProfile | undefined> {
 		const defaultProvider = this.getDefaultProviderName();
+		// Handle unsupported provider
+		if (model?.providerName && !this._capabilitiesService.providers[model.providerName]) {
+			const installed = await this._connectionManagementService.handleUnsupportedProvider(model.providerName);
+			if (!installed) {
+				// User cancelled install prompt so exit early since we won't be able to connect
```

Comment:
What is the behavior with this - does the dialog just not open? Or does it open but is blank? I think my slight preference is the first - although if we do open it I would at least suggest putting in what info we can (and just let the provider default to MSSQL)

Reply:
Dialog doesn't open if user cancels the prompt.

Reply:
Sounds good, that seems like reasonable behavior for now.